File: LeetCode/CodingInterviews/MyPow.java
package LeetCode;

class Solution {
    public double myPow(double x, int n) {
        if (x == 0)
            return 0;
        long b = n;
        if (b < 0) {
            x = 1 / x;
            b = -b;
        }
        double res = 1.0;
        while (b > 0) {
            if ((b & 1) == 1)
                res *= x;
            x *= x;
            b >>= 1;
        }
        return res;
    }
}

File: LeetCode/CodingInterviews/Go/myPow.go
package coding

func myPow(x float64, n int) float64 {
	if n < 0 {
		n = -n
		x = 1.0 / x
	}
	var res float64 = 1.0
	for n != 0 {
		if n&1 == 1 {
			res = res * x
		}
		x = x * x
		n = n >> 1
	}
	return res
}
